src/Player/Player.cpp

```diff
@@ -34,6 +35,111 @@ namespace Player {
 
     void Player::fortify() {
         Terminal::debug("Player fortify");
+        Terminal::print("Fortification phase");
+        vector<string> selections;
+        bool validChoice = false;
+        int source_country_index;
+        Country * source_country;
+
+        bool can_fortify = false;
+        // confirm fortification is possible
+        for (auto & country : *countries) {
+            for (auto & neighbor : *country->get_neighbors()) {
+                if (this == neighbor->get_owner() && country->get_armies() > 1) {
+                    can_fortify = true;
+                    break;
+                }
+            }
+            if (can_fortify) {
+                break;
+            }
+        }
+
+        if (!can_fortify) {
+            Terminal::print("Fortification is not possible, you do not own two valid neighboring countries");
+            return;
+        } else {
```

Comment:
you can remove this else since you return in the previous block

Reply:
good catch, done